django/thunderstore/api/cyberstorm/views/package_versions.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework import serializers
+from rest_framework.generics import ListAPIView, get_object_or_404
+
+from thunderstore.api.utils import CyberstormAutoSchemaMixin
+from thunderstore.repository.models import Package
+
+
+class CyberstormPackageVersionSerializer(serializers.Serializer):
+    version_number = serializers.CharField()
+    datetime_created = serializers.DateTimeField(source="date_created")
+    download_count = serializers.IntegerField(min_value=0, source="downloads")
+    download_url = serializers.CharField(source="full_download_url")
+    install_url = serializers.CharField()
+
+
+class PackageVersionsAPIView(CyberstormAutoSchemaMixin, ListAPIView):
+    """
+    Return a list of available versions of the package.
+    """
+
+    serializer_class = CyberstormPackageVersionSerializer
+
+    def get_queryset(self):
+        package = get_object_or_404(
+            Package.objects.active(),
+            namespace__name=self.kwargs["namespace_id"],
+            name__iexact=self.kwargs["package_name"],
+        )
+
+        return package.versions.active()
```

Comment:
Why do we query the package first instead of just returning a `PackageVersion` queryset with one query?

Reply:
Because I wanted to raise 404 if the package itself doesn't exist. What would you say is the correct combination of responses for [active package | inactive package] x [ active versions | no active versions]?

Reply:
Right, I suppose that makes sense. Could save a query by checking the post-serialization result length & raising 404 there if it's empty I suppose (calling .count() wouldn't work since that's another query ofc), that'd let us query the `PackageVersion` table directly and use 1 query instead of 2.